### PR TITLE
Add time as type

### PR DESCRIPTION
**Pull request by davecromberge**

### CODE REVIEW

File: src/dqe_fun.erl [hunk 1]
```diff
@@ -10,9 +10,12 @@
 
 -export([init/0, reg/1, lookup/2, list/0, render/1]).
 -export_type([spec/0]).
+
+-ignore_xref([behaviour_info/1, list/0, lookup/2, reg/1, render/1]).
+
 -define(TBL, dqe_fun).
 
--type return_type() :: histogram | metric.
+-type return_type() :: histogram | metric | time.
```
Comment (Licenser) [inline src/dqe_fun.erl]: time is not a return type it's a type (see two lines down)

Comment (davecromberge) [inline src/dqe_fun.erl]: I originally thought the same, but the change I made in the parser used `time` as the return:

```

time(T, U) ->
    #{
      op => time,
      args => [T, U],
      signature => [integer, time_unit],
      return => time
     }.
```

Does the return actually refer to the type itself then?

Comment (Licenser) [inline src/dqe_fun.erl]: return type in dqe_fun is the **function** return type, that is something entirely different from the operation map you link there.

from the POV of dqe_fun time is an argument, like integer or float (two lines down). A function can't return a time.

**Comment by Licenser** [comment]
awesoem!

File: src/dqe_fun.erl (hunk 1 above)
Comment (davecromberge) [inline src/dqe_fun.erl]: @Licenser right, functions as data makes sense.

**Comment by davecromberge** [comment]
Thanks Heinz, will wait for a new version and then do a PR for dqe.

**Comment by Licenser** [comment]
already done ;)

**Pull request merged**